### PR TITLE
[breaking] remove importants and shorthand form border rules, closes #864

### LINKED ISSUE #864: Using media query variants with border classes results in unexpected colors
When using `border-l-mm`, this property is applied: `border-left: 1px solid !important;` Because this is using shorthand syntax, it implicitly sets the `border-left-color` value to `initial`, which can't overriden by a `border-color` class (specifically `border--gray-light` in my example). Both classes are using `!important`, but `border--gray-light` class presumably comes before the `border-l-mm` class in the source.

One solution might be to change the `border-l` family of classes to specifically apply the `border-width` and `border-style`, though this would increase the file size. Or we could alter the source order. Any thoughts on this?

The class list I'm using is: `border-t border-t--0-mm border-l-mm border--gray-light`

And here's the result:
screen shot 2017-10-03 at 11 18 49 am
screen shot 2017-10-03 at 11 18 39 am

### PR DESCRIPTION
**Pull request by samanpwbb**
@davidtheclark for review, cc: @jfurrow

### CODE REVIEW

**Review by samanpwbb — COMMENTED** [review]
when border--2 is used now, it overrides directional border rules. Hm. I think we need to add specificity to solve this problem. Changes incoming.

**Comment by samanpwbb** [comment]
Okay: I had to make breaking changes to border--2 but I think it's the best option.

File: src/theming.css [hunk 1]
```diff
@@ -20,11 +20,26 @@
...
+  border-style: solid;
+  border-right-width: 1px;
+}
+.border-b {
+  border-style: solid;
+  border-bottom-width: 1px;
+}
+.border-l {
+  border-style: solid;
+  border-left: 1px solid;
```
Comment (davidtheclark) [inline src/theming.css]: `border-left-width: 1px`

**Review by davidtheclark — APPROVED** [review]
🎖 one change to make before merging.

**Comment by jfurrow** [comment]
🎉  Thanks for this, it looks great!

**Pull request merged**